lib/build.ts

```diff
@@ -1,344 +1,384 @@
 /**
  * Script to handle the build process for creating the Managed App zip file
- * 
+ *
  * @author Russell Seymour
  */
 
 // Libraries --------------------------------------------------------------
 import * as program from "commander";
-import * as path from "path";
-import * as fs from "fs-extra";
-import * as rimraf from "rimraf";
-import {sprintf} from "sprintf-js"
+import {copySync, ensureDirSync, existsSync, readFileSync, statSync, unlinkSync, writeFileSync} from "fs-extra";
+import {basename, isAbsolute, join as pathJoin, resolve} from "path";
+import {sync as rimrafSync} from "rimraf";
+import {sprintf} from "sprintf-js";
 import * as zip from "zip-folder";
 
+// Define constants for the keys in the buildConfig object
+const dirsKey = "dirs";
+const appRootKey = "appRoot";
+const outputKey = "output";
+const workingKey = "working";
+const buildKey = "build";
+const productionKey = "production";
+const stagingKey = "staging";
+const filesKey = "files";
+const functionsKey = "functions";
+const templateFileKey = "template_file";
+const configKey = "config";
+const codeKey = "code";
+const codeFilesKey = "code_files";
+const templateKey = "template";
+const variablesKey = "variables";
+const propertiesKey = "properties";
+const resourcesKey = "resources";
+const parametersKey = "parameters";
+const baseUrlKey = "baseUrl";
+const defaultValueKey = "defaultValue";
+const outputsKey = "outputs";
+const verifyURLApiKeyKey = "verifyURLApiKey";
+const packageKey = "package";
+const nameKey = "name";
```

Comment:
👏  Happy to see these. I don't think the `...Key` suffix in var names adds anything though and can be removed

Reply:
I put these in so that there could be no clash with other variables in the code. For example I think that `outputs` is a variable used elsewhere.